### PR TITLE
Display incomplete but non-actionable tasks in case timeline PR

### LINKED ISSUE #8220: Display incomplete but non-actionable tasks in case timeline
image

## Acceptance criteria
* [ ] Display incomplete but non-actionable tasks in case timeline

### PR DESCRIPTION
**Pull request by rkreyhsig**
Resolves #8220 

nonactionableincasetimeline

| Before | After |
| --- | --- |
|screen shot 2019-01-23 at 1 24 53 pm|screen shot 2019-01-23 at 1 22 18 pm|

### CODE REVIEW

**Comment by rkreyhsig** [comment]
Per @lowellrex and @eric-nava we are moving these tasks to the TaskSnapshot

| Before | After |
| --- | --- |
|screen shot 2019-01-25 at 11 43 39 am|tasksnapshotafter|

File: client/app/queue/components/TaskRows.jsx [hunk 1]
```diff
@@ -14,6 +14,8 @@ import CaseDetailsDescriptionList from '../components/CaseDetailsDescriptionList
 import CO_LOCATED_ADMIN_ACTIONS from '../../../constants/CO_LOCATED_ADMIN_ACTIONS.json';
 import ActionsDropdown from '../components/ActionsDropdown';
 import OnHoldLabel from '../components/OnHoldLabel';
+import { incompleteNonActionableTasks } from '../selectors';
```
Comment (lowellrex) [inline client/app/queue/components/TaskRows.jsx]: Are we importing the `incompleteNonActionableTasks` for any reason other than to show/hide instructions for individual tasks?

File: client/app/queue/components/TaskRows.jsx [hunk 2]
```diff
@@ -63,18 +65,32 @@ const taskInfoWithIconTimelineContainer =
...
 class TaskRows extends React.PureComponent {
   constructor(props) {
     super(props);
+    const taskVisibiltyObj = {};
+
+    this.props.incompleteNonActionableTasks.forEach((task) => taskVisibiltyObj[task.uniqueId] = false);
+
     this.state = {
-      taskInstructionsIsVisible: false
+      taskInstructionsIsVisible: false,
```
Comment (lowellrex) [inline client/app/queue/components/TaskRows.jsx]: Let's structure this like:
```jsx
this.state = {
  taskInstructionsIsVisible: {
    "task_id_1": true,
    "task_id_2": false,
    "task_id_3": true,
  }
}
```
with the initial state being
```jsx
this.state = {
  taskInstructionsIsVisible: { }
}
```

This allows us to not have to pull in the set of appeals we will be showing in the Component's constructor.

File: client/app/queue/components/TaskRows.jsx (hunk 1 above)
Comment (rkreyhsig) [reply client/app/queue/components/TaskRows.jsx]: I've already gotten rid of that, just haven't pushed it...

File: client/app/queue/components/TaskRows.jsx [hunk 3]
```diff
@@ -240,7 +260,8 @@ class TaskRows extends React.PureComponent {
               { this.taskInstructionsListItem(task) }
             </CaseDetailsDescriptionList>
           </td>
-          { !timeline && <td {...taskActionsContainerStyling}> { this.showActionsListItem(task, appeal) } </td> }
+          { !timeline && <td {...taskActionsContainerStyling}>
+            { task.availableActions.length > 0 && this.showActionsListItem(task, appeal) } </td> }
```
Comment (lowellrex) [inline client/app/queue/components/TaskRows.jsx]: Let's encapsulate the `task.availableActions.length > 0` logic in the `showActionsListItem` function.

Comment (rkreyhsig) [reply client/app/queue/components/TaskRows.jsx]: ✅

File: client/app/queue/components/TaskRows.jsx (hunk 2 above)
Comment (rkreyhsig) [reply client/app/queue/components/TaskRows.jsx]: Do you mean not having to pull in the set of tasks?  Not sure I'm following how we can build the hash with the id's without bringing in the taskList...

File: client/app/queue/TaskSnapshot.jsx [hunk 4]
```diff
@@ -64,11 +66,15 @@ export class TaskSnapshot extends React.PureComponent<Props> {
 
 const mapStateToProps = (state: State, ownProps: Params) => {
   const { userRole } = state.ui;
+  let nonActionableTasks = incompleteNonActionableTasks(state, { appealId: ownProps.appealId });
+
+  nonActionableTasks = _.orderBy(nonActionableTasks, ['createdAt'], ['desc']);
```
Comment (lowellrex) [inline client/app/queue/TaskSnapshot.jsx]: Let's sort these tasks [in the selector](https://github.com/department-of-veterans-affairs/caseflow/pull/8855/files#diff-2a3f4e296b35f6015a2d1d3ef72e5506R216).

File: client/app/queue/components/TaskRows.jsx (hunk 2 above)
Comment (lowellrex) [reply client/app/queue/components/TaskRows.jsx]: Right. Let's chat about this after lunch

**Pull request merged**